pkg/netflow/portrollup/portrollup.go

```diff
@@ -137,9 +139,36 @@ func (prs *EndpointPairPortRollupStore) UseNewStoreAsCurrentStore() {
 	defer prs.mu.Unlock()
 
 	prs.curStore = prs.newStore
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** --Not to be addressed in this PR-- ‏Here, the previous `curStore` will need to be garbage collected. It can contains a significant amount of data, so it might be interesting (if not already done) to analyse the performance penalty caused by this GC phase.

Reply:
@gmarabout Indeed, you are right. I did some analysis and when curStore is delete/overwritten, there is indeed a lot of heap to be garbage collected. This is very likely the cause of the drop in messages we are seeing in some cases.

The ways to mitigate this are:
- allocated less memory (this PR and https://github.com/DataDog/datadog-agent/pull/13792)
- use larger buffer by default https://github.com/DataDog/datadog-agent/pull/13755 and possibly recommend users to use larger buffer when needed, we will be able see when it's required with the new telemetry https://github.com/DataDog/datadog-agent/pull/13756

There are possibly future improvements like:
- find ways to smooth out GC
- maybe newer versions of go might also improve things
- different port rollup tracker implementation

But for now, I think with the 2 mitigations mentioned above should already help. We can first test them in real condition and see if there is a need to further improvement things.